fuzz/filter/filter.go

```diff
@@ -90,8 +91,8 @@ func ApplyFilters(filters []Filter, root token.Token) (token.Token, error) {
 					return nil, fmt.Errorf("error in fuzzing filter %v: %s", filters[i], err)
 				}
 
-				if len(replacement) > 0 {
-					newTokens = append(newTokens, replacement...)
+				if replacement != nil {
```

Comment:
I took another look at this. I think my original code is wrong. I think what I really intended to do is the following:
- For every token loop through the filters
- Stop at the first filter that has a replacement
- Replace the original token
- Continue to the next token
- Do not replace tokens which where replaced by filters <- this is handled by the `known` map

Could you please change the code and the method documentation accordingly.